src/libraries/System.Formats.Tar/src/System/Formats/Tar/PosixTarEntry.cs

```diff
@@ -62,7 +62,10 @@ public int DeviceMajor
                 }
 
                 ArgumentOutOfRangeException.ThrowIfNegative(value);
-                ArgumentOutOfRangeException.ThrowIfGreaterThan(value, 0x1FFFFF); // 7777777 in octal
+                if (FormatIsOctalOnly)
```

Comment:
Just to be clear that I get this: Are DeviceMajor and DeviceMinor values greater than 0x1FFFFF valid in Unix?

Reply:
Yes. Unix/POSIX says that you can turn a `dev_t` into a major and a minor and vice versa. It doesn't put a limit on the size of `dev_t` or say how many bits are used by the major and how many by the minor.

You can look at a concrete implementation (like a Linux kernel) and determine what its limits are, but there is no standard that sets an upper bound.